### DISCUSSION TITLE
Inquiry about Identifying the Compiler Used to Install WWIII Model on HPC

### DISCUSSION THREAD

**Discussion by krishnanathira** [NOAA-EMC/WW3#1140] [Q&A] [answered]
Hi everyone,

I have a question regarding the installation of the WWIII model on our HPC system. There is already a WWIII model installed in the HPC account, and I am attempting to replicate the installation in my own account.

Specifically, I am seeking guidance on how to determine the compiler that was used to install the existing WWIII model. Knowing the compiler information is crucial for ensuring compatibility and replicating the setup accurately.

Could someone kindly provide insights or steps on how I can identify the compiler used for the current WWIII model installation? Your assistance is highly appreciated.

Thank you,
Athira

**Comment by JessicaMeixner-NOAA** [comment]
Have you reached out to who installed WW3 on your HPC system?

**Reply by krishnanathira** [reply]
I couldn't actually.
Thanks

**Comment by ukmo-ccbunney** [comment, marked as answer]
Sometimes the compiler stores its name and version in the executable. You can access this information using `objdump` or `nm`, e.g.

`objdump -h your_exe`

The `-h` flag displays the headers in the executable. The compiler version is sometimes stored in the `.note` or `.comment` headers. However, often this info is stripped out when doing a release build (without -g flags)...

It's worth a try though!

**Reply by krishnanathira** [reply]
Thanks a lot.
I will certainly try this.

**Comment by hot007** [comment]
To add to the `objdump` suggestion, I also use `ldd your_exectuable` to get a list of the libraries linked. Whether the output is useful depends on how the HPC's module system and compiler set up is designed but often it can be helpful.

**Reply by krishnanathira** [reply]
Thanks a lot

**Reply by CarstenHansen** [reply]
On our setup we require that the execution environment matches the build environment. To control that I set up the right execution environment, I run `readelf -d ww3_shel`. The output is not as detailed as `ldd your_exectuable`, but provides all the library runpaths in one line, e.g.:

    0x000000000000001d (RUNPATH)            Library runpath: [/opt/netcdf-4.9.2/gcc-netcdf4/lib:/opt/netcdf-fortran-4.6.0/gfortran-13.1.0-ieee/lib:/opt/gcc-13.1.0/lib64:/opt/isl-0.24/lib:/opt/mpc-1.3.1/lib:/opt/mpi/openmpi-4.1.5/gfortran-13.1.0-ieee/lib]`

**Reply by krishnanathira** [reply]
Thats great.
Many Thanks!
